### PR TITLE
AWS: fix integration test after assert library refactor

### PR DESCRIPTION
**Pull request by jackye1995**
@kbendick @yyanyy 

Integration test fails when running 0.12.0 build verification, due to the change in #2721. We never merged this internally so did not get a chance to test it.

### CODE REVIEW

**Review by yyanyy — APPROVED** [review]

**Review by rdblue — APPROVED** [review]

File: api/src/test/java/org/apache/iceberg/AssertHelpers.java [hunk 1]
```diff
@@ -113,6 +113,37 @@ public static void assertThrowsCause(String message,
...
+                                           Class<? extends Exception> expected,
+                                           String expectedContainedInMessage,
+                                           Class<? extends Exception> cause,
+                                           String causeContainedInMessage,
+                                           Runnable runnable) {
+    AbstractThrowableAssert<?, ?> chain = Assertions.assertThatThrownBy(runnable::run)
+        .as(message)
+        .isInstanceOf(expected);
+
+    if (expectedContainedInMessage != null) {
```
Comment (rdblue) [inline api/src/test/java/org/apache/iceberg/AssertHelpers.java:137]: Why the null checks? I don't think we want anyone pass null in to turn off the validation right?

**Comment by rdblue** [comment]
Looks good to me overall. I'd like to get this into 0.12.0 to fix the failure, too.

Out of curiosity, how was this passing before? Wouldn't this have failed before the assert fix?

**Comment by jackye1995** [comment]
@rdblue yeah it should have been breaking since the commit #2721, but I assumed the integration tests were run by Kyle and did not verify it, and we never merged that change to our internal repo to test, so didn't find out the issue until now. 

I think we have been talking about contributing an AWS account to run the integration tests as a part of the CI for any change related to the AWS module. It is currently application in progress, it should be the best way to avoid this kind of issue in the future.

**Comment by kbendick** [comment]
> @rdblue yeah it should have been breaking since the commit #2721, but I assumed the integration tests were run by Kyle and did not verify it, and we never merged that change to our internal repo to test, so didn't find out the issue until now.
> 
> I think we have been talking about contributing an AWS account to run the integration tests as a part of the CI for any change related to the AWS module. It is currently application in progress, it should be the best way to avoid this kind of issue in the future.

Thanks for this fix. I also (mistakenly) assumed that these tests were run during CI.

It would be great if there were a way to run the integration tests on a semi-regular basis, or to add this to CI (even using dynamo-local and friends if need be to do it during CI - though ideally with real infrastructure would be preferable as I know at least dynamo-local isn't quite the same as DynamoDB from past experience). Or, do we have documentation for running these specific tests? I regularly run some test suites against AWS (in addition to other infrastructure) for things like releases, but it's just my own ad-hoc set up for things like ensuring table creation works, etc.

FWIW, I just pulled master and I still get NPE when running `./gradlew :iceberg-aws:integrationTest`. How am I supposed to be running these? When I run `./gradlew clean build`, everything passes. But running `./gradlew :iceberg-aws:integrationTest` gives me several failures.

```
> Task :iceberg-aws:integrationTest

org.apache.iceberg.aws.glue.GlueCatalogNamespaceTest > classMethod FAILED
    java.lang.ExceptionInInitializerError
        Caused by: software.amazon.awssdk.core.exception.SdkClientException

org.apache.iceberg.aws.glue.GlueCatalogNamespaceTest > classMethod FAILED
    java.lang.NoClassDefFoundError

..... (more NoClassDefFoundError).
```

**Comment by kbendick** [comment]
Maybe in the interim, we can add a `README.md` file to the `aws/src/integration` or something? I'm not able to get this to work at all locally - I assume it requires input parameters etc.

At the least, when testing releases etc, I can run the S3 specific subset of these (as I don't use GlueCatalog etc currently, but do make use of S3FileIO).

**Comment by kbendick** [comment]
cc @jackye1995 On the above two questions (sorry for multiple comments).

**Comment by kbendick** [comment]
> Looks good to me overall. I'd like to get this into 0.12.0 to fix the failure, too.
> 
> Out of curiosity, how was this passing before? Wouldn't this have failed before the assert fix?

To answer your question somewhat more directly @rdblue of how this was passing before, this doesn't appear to be run during any of our existing automation. Even running them manually, one has to have the environment set up for it. TIL.

**Pull request merged**